kwave/utils/matlab.py

```diff
@@ -89,6 +90,9 @@ def matlab_mask(arr: np.ndarray, mask: np.ndarray, diff: Optional[int] = None) -
 
     """
 
+    if mask.dtype == "uint8":
```

Comment:
what if we convert uint8 to uint16 and int8 to int16?

Reply:
Exaclty, current "value -1 is out of bounds of uint8 type"

Reply:
```python
>>> x
array([0], dtype=uint8)
>>> x - 1
array([255], dtype=uint8)
>>> np.version
<module 'numpy.version' from '/Users/walters/miniconda3/lib/python3.12/site-packages/numpy/version.py'>
>>> np.__version__
'1.26.4'
```

Here is the legacy behavior

Reply:
Here is the new behavior:

```python
>>> import numpy as np
>>> x = np.array([0], dtype='uint8')
>>> x - 1
array([255], dtype=uint8)
>>> np.__version__
'2.1.3'

```

looks like my assumption was wrong

Reply:
Seems to be an issue with type promotion with the + operator.

```python

>>> import numpy as np
>>> x = np.array([0], dtype='uint8')
>>> x - 1
array([255], dtype=uint8)
>>> x.ravel(order='F') + -1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: Python integer -1 out of bounds for uint8
>>> x.ravel(order='F').astype("int8") + -1
array([-1], dtype=int8)
>>> x + -1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: Python integer -1 out of bounds for uint8
>>> np.__version__
'2.1.3'
```

Reply:
This is all related to this:
https://github.com/numpy/numpy/issues/25639#issuecomment-1901166868